AppTests/Networking/FixedSizeJsonRequestTests.swift

```diff
@@ -0,0 +1,246 @@
+    let request1 = try MockEmptyRequest(targetSize: targetSize, path: "").asURLRequest()
+    let request2 = try MockEmptyRequest(targetSize: targetSize, path: "/some/endpoint").asURLRequest()
+    let request3 = try MockEmptyRequest(targetSize: targetSize, path: "/some/other/endpoint?query=1").asURLRequest()
+
+    XCTAssertEqual(request1.size, targetSize)
+    XCTAssertEqual(request1.size, request2.size)
+    XCTAssertEqual(request1.size, request3.size)
+  }
+
+  func testConcreteRequestsHaveSameSize() throws {
+    let targetSize = 50000
+
+    let teks = (0 ..< 14).map { _ in CodableTemporaryExposureKey.mock() }
+    let summaries = (0 ..< 30).map { _ in CodableExposureDetectionSummary.mock() }
+
+    let request1 = try DataUploadRequest(
+      body: .init(
+        teks: teks,
+        province: "AA",
+        exposureDetectionSummaries: summaries
+      ),
+      otp: OTP(),
+      now: { Date() },
+      targetSize: targetSize
+    ).asURLRequest()
+
+    let request2 = try OTPValidationRequest(otp: OTP(), now: { Date() }, targetSize: targetSize).asURLRequest()
+    let request3 = try DummyIngestionRequest(now: { Date() }, targetSize: targetSize).asURLRequest()
+
+    XCTAssertEqual(request1.size, targetSize)
+    XCTAssertEqual(request1.size, request2.size)
+    XCTAssertEqual(request1.size, request3.size)
+  }
+}
+
+// MARK: - Helpers
+
+private extension URLRequest {
+  var size: Int {
+    let bodySize = self.httpBody?.count ?? 0
+    let headersSize = self.headers.map { "\($0.name): \($0.value)\r\n" }.map { $0.data(using: .utf8)?.count ?? 0 }.reduce(0, +)
+    let methodSize = self.method?.rawValue.data(using: .utf8)?.count ?? 0
+    let urlSize = self.url?.absoluteString.data(using: .utf8)?.count ?? 0
+    return bodySize + headersSize + methodSize + urlSize
+  }
+}
+
+// MARK: - Mocks
+
+extension FixedSizeJsonRequestTests {
+  struct MockRequest: FixedSizeJSONRequest {
+    typealias BodyModel = Body
+
+    var targetSize: Int
+    var method: HTTPMethod
+    var baseURL: URL
+    var path: String
+    var headers: [HTTPHeader]
+
+    let jsonParameter: Body
+
+    init(
+      targetSize: Int,
+      method: HTTPMethod = .post,
+      baseURL: URL = URL(string: "https://example.com")!,
+      path: String = "/some/endpoint",
+      headers: [HTTPHeader] = [],
+      body: Body
+    ) {
+      self.targetSize = targetSize
+      self.method = method
+      self.baseURL = baseURL
+      self.path = path
+      self.headers = headers
+      self.jsonParameter = body
+    }
+
+    struct Body: Encodable {
+      let string: String
+      let int: Int
+      let array: [Bool]
+    }
+  }
+
+  struct MockEmptyRequest: FixedSizeJSONRequest {
+    typealias BodyModel = EmptyBody
+
+    var targetSize: Int
+    var method: HTTPMethod
+    var baseURL: URL
+    var path: String
+    var headers: [HTTPHeader]
+
+    init(
+      targetSize: Int,
+      method: HTTPMethod = .post,
+      baseURL: URL = URL(string: "https://example.com")!,
```

Comment:
<!--
  0 failure: 
  1 warning:  swiftlint (0.39.2...
  
  
  DangerID: danger-id-Danger;
  File: AppTests/Networking/FixedSizeJsonRequestTests.swift;
  Line: 178;
-->

- :warning: swiftlint (0.39.2): Force unwrapping should be avoided. (force_unwrapping)